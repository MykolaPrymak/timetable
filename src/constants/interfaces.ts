type DayOfWeek = 'monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday';

export const DailyTimeTableTypeOpen = 'open';
export const DailyTimeTableTypeClose = 'close';

export type DailyTimeTableType = typeof DailyTimeTableTypeOpen | typeof DailyTimeTableTypeClose;

export interface DailyTimeTable {
  type: DailyTimeTableType;
  value: number;
}

export interface WeekDayItem {
  title: string;
  key: DayOfWeek;
}

export type TimeTable = {
  [key in DayOfWeek]?: DailyTimeTable[];
};
