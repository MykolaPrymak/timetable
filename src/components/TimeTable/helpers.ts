import {
  TimeTable as TimeTableType,
  DayOfWeek,
  DailyTimeTable,
  DailyTimeTableType,
  DailyTimeTableTypeOpen,
  DailyTimeTableTypeClose,
} from '../../constants/interfaces';

// Return empty or sorted day
export const getDailyTimeTable = (dayOfWeek: DayOfWeek, timetable: TimeTableType): DailyTimeTable[] => {
  const dayTimeTable = timetable[dayOfWeek] || [];

  // Make array copy and then return it
  return dayTimeTable.slice().sort((a, b) => a.value - b.value);
};

export const formatTime = (time: number = 0): string => {
  // Normalize the value
  // and convert to ms for Date constructor
  time = (time % 86400) * 1000;
  const date = new Date(time);

  // We use UTC functions because Date constructor accepts seconds in UTC
  let hours = date.getUTCHours();

  const isAM = hours < 12;
  if (hours === 0) {
    hours = 12;
  } else if (hours > 13) {
    hours -= 12;
  }
  const minutes = date.getUTCMinutes();
  const seconds = date.getUTCSeconds();

  const displayValues = [hours];
  if (minutes > 0 || seconds > 0) {
    displayValues.push(minutes);

    if (seconds > 0) {
      displayValues.push(seconds);
    }
  }

  // Add leading "0" if need, join by ":" and add AM/PM
  return `${displayValues.map((val, idx) => `${idx > 0 ? '0' : ''}${val}`.slice(-2)).join(':')} ${isAM ? 'AM' : 'PM'}`;
};

export const getNextWeekDayKey = (dayOfWeek: DayOfWeek): DayOfWeek => {
  const weekDays: DayOfWeek[] = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'];
  let weekDayIdx = weekDays.indexOf(dayOfWeek);

  if (weekDayIdx === weekDays.length - 1) {
    weekDayIdx = -1;
  }

  // Go one day further and return it's key
  return weekDays[++weekDayIdx];
};

const getTimeIdx = (type: DailyTimeTableType, dayTimeTable: DailyTimeTable[], startIdx: number = 0) => {
  // Don't allow negative startIdx
  startIdx = Math.max(0, startIdx);
  const foundedIdx = dayTimeTable.slice(startIdx).findIndex(timeRecord => timeRecord.type === type);

  // Add startIdx to found index to respect original array indexes
  // If item not found - return -1
  return foundedIdx === -1 ? foundedIdx : foundedIdx + startIdx;
};

export const getOpenTimeIdx = getTimeIdx.bind({}, DailyTimeTableTypeOpen);
export const getCloseTimeIdx = getTimeIdx.bind({}, DailyTimeTableTypeClose);
