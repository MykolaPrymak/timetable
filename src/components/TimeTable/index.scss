@import 'styles/colors.scss';

.TimeTable {
  font-family: 'Roboto', sans-serif;
  font-weight: normal;
  text-align: left;
  background-color: $white-color;

  position: absolute;
  top: 100px;
  left: 100px;
  
  width: 400px;
  height: 600px;
}

.TimeTable-header {
  /*
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  */
}

.TimeTable-label {
  color: #61dafb;
}
