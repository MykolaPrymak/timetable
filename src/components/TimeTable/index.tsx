import AccessTimeIcon from '@material-ui/icons/AccessTime';
import { TimeTable as TimeTableType, WeekDayItem } from '../../constants/interfaces';
import './index.scss';

//
const WEEK_DAYS: WeekDayItem[] = [
  {
    title: 'Monday',
    key: 'monday',
  },
  {
    title: 'Tuesday',
    key: 'tuesday',
  },
  {
    title: 'Wednesday',
    key: 'wednesday',
  },
  {
    title: 'Thursday',
    key: 'thursday',
  },
  {
    title: 'Friday',
    key: 'friday',
  },
  {
    title: 'Saturday',
    key: 'saturday',
  },
  {
    title: 'Sunday',
    key: 'sunday',
  },
];

const TimeTableItem = (props: { weekDay: WeekDayItem; timetable: TimeTableType; isToday: boolean }) => {
  const { weekDay, isToday } = props;
  // Sort current day times
  // Find open one - if not - the day is Closed
  // if today - show label

  return (
    <div className="TimeTable-weekday">
      <div className="TimeTable-weekday-name">
        {weekDay.title}
        {isToday && <label className="TimeTable-label">Today</label>}
      </div>
      <div className="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none">Closed</div>
    </div>
  );
};

const TimeTable = (props: { timetable: TimeTableType }) => {
  const { timetable } = props;
  // Get current day number 0-6 to determine today label
  // Because 0 represents Sunday we need to do shift
  let currentDayIdx = new Date().getDay() - 1;
  if (currentDayIdx === -1) {
    currentDayIdx = 6;
  }

  return (
    <div className="TimeTable">
      <h4 className="TimeTable-header">
        <AccessTimeIcon className="TimeTable-icon" />
        Opening hours
      </h4>
      <div className="TimeTable-weekdays">
        {WEEK_DAYS.map((weekDay, idx) => (
          <TimeTableItem key={weekDay.key} weekDay={weekDay} isToday={idx === currentDayIdx} timetable={timetable} />
        ))}
      </div>
    </div>
  );
};

export default TimeTable;
