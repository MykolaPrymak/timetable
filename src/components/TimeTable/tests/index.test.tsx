import { render } from '@testing-library/react';

import { TimeTable as TimeTableType } from '../../../constants/interfaces';
import TimeTable from '../index';

// Open just one day in the week
const oneDayTimeTable: TimeTableType = {
  monday: [
    { type: 'open', value: 32400 },
    { type: 'close', value: 72000 },
  ],
};

// Opens only one day and closed on next
const nextDayCloseTimeTable: TimeTableType = {
  monday: [
    {
      type: 'close',
      value: 3600,
    },
  ],
  sunday: [
    {
      type: 'open',
      value: 64800,
    },
  ],
};

// Closing on next day and have next day time slots
const nextDayCloseAndOpenTimeTable: TimeTableType = {
  friday: [
    {
      type: 'open',
      value: 64800,
    },
  ],
  saturday: [
    {
      type: 'close',
      value: 3600,
    },
    {
      type: 'open',
      value: 32400,
    },
    {
      type: 'close',
      value: 39600,
    },
    {
      type: 'open',
      value: 57600,
    },
    {
      type: 'close',
      value: 82800,
    },
  ],
};

// Full data formatting example
const fullWeekTimeTable: TimeTableType = {
  monday: [],
  tuesday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  wednesday: [],
  thursday: [
    { type: 'open', value: 36000 },
    { type: 'close', value: 64800 },
  ],
  friday: [{ type: 'open', value: 36000 }],
  saturday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 36000 },
  ],
  sunday: [
    { type: 'close', value: 3600 },
    { type: 'open', value: 43200 },
    { type: 'close', value: 75600 },
  ],
};

// Multislot day
const multislotWeekTimeTable: TimeTableType = {
  monday: [],
  tuesday: [
    { type: 'open', value: 3600 },
    { type: 'close', value: 7200 },
    { type: 'open', value: 36000 },
    { type: 'close', value: 39800 },
    { type: 'open', value: 43200 },
    { type: 'close', value: 75600 },
  ],
};

describe('<TimeTable />', () => {
  let getDay = Date.prototype.getDay;

  beforeEach(() => {
    // Mock new Date().getDay() to prevent changing the today label
    Date.prototype.getDay = () => 6; // We will have always Saturday
  });

  afterAll(() => {
    Date.prototype.getDay = getDay;
  });

  it('renders default all closed timetable if nothing is passed', () => {
    const { container } = render(<TimeTable />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders today label at Sunday', () => {
    Date.prototype.getDay = () => 0;
    const { container } = render(<TimeTable />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders one open day in the week', () => {
    const { container } = render(<TimeTable timetable={oneDayTimeTable} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders properly if close time is on next day', () => {
    const { container } = render(<TimeTable timetable={nextDayCloseTimeTable} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders properly if close time is on next day and we have time slots on next day', () => {
    const { container } = render(<TimeTable timetable={nextDayCloseAndOpenTimeTable} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders properly if we have full week of time slots', () => {
    const { container } = render(<TimeTable timetable={fullWeekTimeTable} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders properly if we have couple of time slots in one day', () => {
    const { container } = render(<TimeTable timetable={multislotWeekTimeTable} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
