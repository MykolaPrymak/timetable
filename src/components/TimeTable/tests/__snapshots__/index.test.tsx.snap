// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeTable /> renders default all closed timetable if nothing is passed 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders one open day in the week 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            9 AM
          </span>
           - 
          <span>
            8 PM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders properly if close time is on next day 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            6 PM
          </span>
           - 
          <span>
            1 AM
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders properly if close time is on next day and we have time slots on next day 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            6 PM
          </span>
           - 
          <span>
            1 AM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            9 AM
          </span>
           - 
          <span>
            11 AM
          </span>
        </span>
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            4 PM
          </span>
           - 
          <span>
            11 PM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders properly if we have couple of time slots in one day 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            1 AM
          </span>
           - 
          <span>
            2 AM
          </span>
        </span>
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            10 AM
          </span>
           - 
          <span>
            11:03:20 AM
          </span>
        </span>
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            12 PM
          </span>
           - 
          <span>
            9 PM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders properly if we have full week of time slots 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            10 AM
          </span>
           - 
          <span>
            6 PM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            10 AM
          </span>
           - 
          <span>
            6 PM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            10 AM
          </span>
           - 
          <span>
            1 AM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            10 AM
          </span>
           - 
          <span>
            1 AM
          </span>
        </span>
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
      </div>
      <div
        class="TimeTable-weekday-workingHours"
      >
        <span
          class="TimeTable-timeSlot"
        >
          <span>
            12 PM
          </span>
           - 
          <span>
            9 PM
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TimeTable /> renders today label at Sunday 1`] = `
<div
  class="TimeTable"
>
  <h4
    class="TimeTable-header"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root TimeTable-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    Opening hours
  </h4>
  <div
    class="TimeTable-weekdays"
  >
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Monday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Tuesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Wednesday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Thursday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Friday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Saturday
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
    <div
      class="TimeTable-weekday"
    >
      <div
        class="TimeTable-weekday-name"
      >
        Sunday
        <label
          class="TimeTable-label"
        >
          Today
        </label>
      </div>
      <div
        class="TimeTable-weekday-workingHours TimeTable-weekday-workingHours--none"
      >
        Closed
      </div>
    </div>
  </div>
</div>
`;
